plugins/modules/zos_mount.py

```diff
@@ -0,0 +1,995 @@
+        if mp_exists is False:
+            os.mkdir(path)
+        mp_exists = os.path.exists(path)
+        if mp_exists is False:
+            module.fail_json(
+                msg="Mount destination (" + path + ") doesn't exist",
+                stderr=str(res_args)
+            )
+
+# Need to see if mountpoint is in use for idempotence
+    currently_mounted = False
+    rc, stdout, stderr = module.run_command(
+        'df | grep ' + src + ' | wc -m', use_unsafe_shell=True)
+    if rc != 0:
+        module.fail_json(
+            msg="Checking for source (" + src + ") failed with error",
+            stderr=str(res_args)
+        )
+    tmpint = int(stdout)
+    if tmpint != 0:  # zero bytes means not found
+        currently_mounted = True
+
+
+# can type be validated?
+
+    # ##########################################
+    # Assemble the mount command
+
+    parmtext = '/* VVV zos_mount VVV '
+    if(len(tabcomment) > 2):
+        parmtext = parmtext + tabcomment[:21]
+    d = datetime.today()
+    parmtext = parmtext + d.ctime()[4:] + " VVV */\n"
+    parmtail = "\n" + parmtext.replace('V', 'A')
+
+    fullcmd = ''
+    fullumcmd = ''
+
+    if gonna_mount:
+        fullcmd = "tsocmd MOUNT FILESYSTEM\\( \\'{0}\\' \\) MOUNTPOINT\\( \\'{1}\\' \\) TYPE\\( '{2}' \\)".format(
+            src, path, fstype)
+        parmtext = parmtext + \
+            "MOUNT FILESYSTEM('{0}')\n      MOUNTPOINT('{1}')\n      TYPE('{2}')".format(
+                src, path, fstype)
+        if 'ro' in opts or 'RO' in opts:
+            subcmd = 'READ'
+        else:
+            subcmd = 'RDWR'
+        fullcmd = "{0} MODE\\({1}\\)".format(fullcmd, subcmd)
+        parmtext = "{0}\n      MODE({1})".format(parmtext, subcmd)
+
+        if len(src_params) > 1:
+            fullcmd = "{0} PARM\\(\\'{1}\\'\\)".format(fullcmd, src_params)
+            parmtext = "{0}\n      PARM('{1}')".format(parmtext, src_params)
+
+        if len(tag_flag) > 0:
+            fullcmd = "{0} TAG\\({1},{2}\\)".format(
+                fullcmd, tag_flag, tag_ccsid)
+            parmtext = "{0}\n      TAG({1},{2})".format(
+                parmtext, tag_flag, tag_ccsid)
+
+        if allow_uids:
+            fullcmd = fullcmd + ' SETUID'
+            parmtext = parmtext + '\n      SETUID'
+        else:
+            fullcmd = fullcmd + ' NOSETUID'
+            parmtext = parmtext + '\n      NOSETUID'
+
+        if 'NOWAIT' in opts or 'nowait' in opts:
+            fullcmd = fullcmd + ' NOWAIT'
+            parmtext = parmtext + '\n      NOWAIT'
+        else:
+            fullcmd = fullcmd + ' WAIT'
+            parmtext = parmtext + '\n      WAIT'
+
+        if 'NOSECURITY' in opts or 'nosecurity' in opts:
+            fullcmd = fullcmd + ' NOSECURITY'
+            parmtext = parmtext + '\n      NOSECURITY'
+        else:
+            fullcmd = fullcmd + ' SECURITY'
+            parmtext = parmtext + '\n      SECURITY'
+
+        if len(sysname) > 1:
```

Comment:
Similar to an earlier comment, may want to verify `sysname` is not `None` before checking length

Reply:
In case I miss any, you will want to ensure any arguments are not `None` when dealing with string parameters that are not required with no default value

Reply:
done